### DISCUSSION TITLE
Assessing Medspacy

### DISCUSSION THREAD

**Discussion by standenman** [medspacy/medspacy#235] [Q&A] [answered]
I am building an application that will take a pdf of medical records and extract key data.   I want the basics: who is the patient, who is the doctor, what is wrong with the patient (diagnosis), key diagnostic test results (MRI show "severe stenosis"), medications, symptoms, patient compliance, patient functional limitations, and more.

MEdspacy seems a logical test, and I just ran a little test on a small text.  The results are disappointing, and I am wondering if I am doing something wrong.   here is my code: 
"import sys
import spacy
import medspacy
import en_core_web_sm

from medspacy.visualization import visualize_ent, visualize_dep
nlp = medspacy.load('en_core_web_sm')
nlp.disable_pipe("parser")
text="""
Stanley F. Denman is a 60 y.o. male with history of fatigue for 10 years, placed on testosterone in 2020 and developed severe substernal chest pain, drove himself to Doctor's hospital and Dr. Mamood Ali placed LAD stent. Unforturnately, he had LVEF 25%, followed and had recurrent chest cramping, had another cardiac cath showing stent in the posterior artery. After his second stent he had ICD placed October 2021. He was told he had CHF at that time. He wants second opinion about his prognosis and his treatment plan. Currently he has severe DOE; can walk 1/2 block with his dog but has severe shortness of breath and heart palpitations- hard heart beating. He agrees to take his medications and has been compliant. After ICD he had chest pain and swelling. He had to be diuresed prior to receiving the ICD. He has gained 30# since MI and is not on 
![Screenshot (71)](https://github.com/medspacy/medspacy/assets/17820082/dff06d77-484d-44dc-891e-d2c1d737f679)
testosterone. He is not on GDMT at this time. He is very upset and it is difficult for him to hear my discussion. Has severe muscle aching in hands and feet on lipitor. Continues to have pain on crestor. Albuterol HFA inhaler 90 mcg/actuation INHALE TWO PUFFS BY MOUTH EVERY 6 HOURS AS NEEDED FOR WHEEZING (Patient taking differently: prn.)
"""
doc = nlp(text)

visualize_ent(doc)
visualize_dep(doc)"

Lots of misses here.  Medications missed. No basic medical tests picked up.   In this sentence, full of vital medical information (LVEF, cardiac cath, stent), "Unforturnately, he had LVEF 25% PERCENT , followed and had recurrent chest cramping, had another cardiac cath showing stent in the posterior artery." we only the percentage tagged.   The negation works for sentence "He is not  on GDMT at this time" but GDMT is a medical abbreviation for "guideline-directed medical therapy"  I thought medspacy dealt with medical abbreviations?

I am not being critical - just trying to get the lay of the land.  Medspacy does not seem to have any advantage in this medical record project of mine over spacy.  Tell me what I am missing?

**Comment by abchapman93** [comment, marked as answer]
Hi @standenman ,
Thanks for your interest in medspaCy! Sounds like a great project.

The model you're using, `en_core_web_sm`, is spaCy's default model, so it's just extracting entities based on a general domain schema. medspaCy mainly adds additional components for additional processing (e.g., negation or section detection) on top of entity extraction that can be done using other trained models or rule-based extraction. We don't really offer pre-trained models in the way that spaCy or scispaCy does.

You could try loading a scispaCy model and then use medspaCy to extract the additional attributes, or create rules using the `TargetMatcher` component. There should be some examples of both in the `notebooks` folder.

**Reply by standenman** [reply]
Thank you for your prompt answer!
